wordpress-mobile/WordPress-iOS
# Migrates notification/push reg remotes to WordPressKit

Fixes #7315 
Fixes #7316 
Fixes #7322 

This PR:
* Migrates the `RemoteNotification` model to the WPKit framework
* Migrates the `RemoteNotificationSettings` model to the WPKit framework
* Migrates the `NotificationSettingsServiceRemote` to the WPKit framework
* Migrates `NotificationSyncServiceRemote` to the WPKit framework
* Migrates `PushAuthenticationServiceRemote` to the WPKit framework
* Migrates related unit tests + mock data to WPKit framework
* Migrates `NSBundle+VersionNumberHelper.h/.m` to the WPShared framework
* Migrates `UIDevice+Helpers.h/.m` to the WPShared framework
* Migrates `Dictionary+Helpers.swift` to the WPShared framework
* Migrates related unit tests + mock data to WPShared framework
* Updates `Podfile` to make `UIDeviceIdentifier` pod available to networking pods
* Updates related imports/refs in main WordPress project
* Fixes some misc documentation warnings

Ref #7301

**To test:**
1. Clean DerivedData.
2. Update CocoaPods repo with pod repo update.
3. Install the pods with pod install.
4. Run _all_ unit tests and make sure they are green.
5. Build & run the app:
a. verify push registration is happening correctly
b. verify all things notification work
